netebpfext/net_ebpf_ext.c

```diff
@@ -34,6 +34,13 @@ static bool _net_ebpf_xdp_providers_registered = false;
 static bool _net_ebpf_bind_providers_registered = false;
 static bool _net_ebpf_sock_addr_providers_registered = false;
 static bool _net_ebpf_sock_ops_providers_registered = false;
+#ifdef _DEBUG
+// Global objects used to store filter contexts that are being cleaned up. This is currently only used in debug
+// contexts.
+EX_SPIN_LOCK _net_ebpf_filter_zombie_list_lock = {0};
+static bool _net_ebpf_filter_zombie_list_initialized = false;
```

Comment:
nit: do we need this?

Reply:
Please correct me if I am wrong - my understanding is that each provider is serialized within calls to itself, but across providers, we have no serialization. Therefore, we could have two providers attaching/detaching at the same time, which could presumably cause issue here, right?

Regarding initialized - is it ever possible to hit the uninitialized codepath and then initialize again after? Now that I am thinking about it, I do not think this may be needed.

Reply:
yes we would need the lock; my comment was only about the `initialized` flag.